extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -578,3 +578,38 @@ export async function getFoldersInFolder(folderPath: string, ignoreBinObj?: bool
 		return await glob(folderFilter, { onlyDirectories: true });
 	}
 }
+
+/**
+ * Determines if provided value is a well-known database source and therefore is allowed to be sent in telemetry
+ * @param value Value to check if it's a well-known database source
+ * @returns Database source value if well-known, otherwise returns undefined
+ */
+export function getWellKnownDatabaseSourceValue(value: string): string | undefined {
+	return constants.WellKnownDatabaseSources
+		.find(wellKnownSource => { return wellKnownSource.toUpperCase() === value.toUpperCase(); });
+}
+
+/**
+ * Get string containing all well-known database sources, separated by semicolon
+ * @returns Well-known database sources, separated by semicolon
+ */
+export function getWellKnownDatabaseSourceString(project: Project): string {
+	let databaseSourceString: string = '';
+	const alreadyAppendedValues = new Set();
+
+	for (let databaseSourceValue of project.getDatabaseSourceValues()) {
+		let wellKnownDatabaseSourceValue = getWellKnownDatabaseSourceValue(databaseSourceValue);
+		if (wellKnownDatabaseSourceValue
+			&& !alreadyAppendedValues.has(wellKnownDatabaseSourceValue)) {
+			if (!databaseSourceString) {
+				databaseSourceString = wellKnownDatabaseSourceValue;
+			} else {
+				databaseSourceString += `;${wellKnownDatabaseSourceValue}`;
```

Comment:
can this be simplified to just adding the wellKnownDatabaseSourceValue to the set, then at the end convert the set to a semicolon separated string?

Reply:
I can, but isn't the code cleaner with only one loop, as opposed to two?

Reply:
why would a second loop be needed? Couldn't it be something like 
```
const databaseSourcesSet = new Set();
for (let databaseSourceValue of project.getDatabaseSourceValues()) {
    const wellKnownDatabaseSourceValue = getWellKnownDatabaseSourceValue(databaseSourceValue);
    if (wellKnownDatabaseSourceValue) {
        databaseSourcesSet.add(wellKnownDatabaseSourceValue);
    }
}
return Array.from(databaseSourcesSet).join(';')
```